services/order.py

```diff
@@ -0,0 +1,29 @@
+from django.db import transaction
+from django.db.models import QuerySet
+from django.contrib.auth import get_user_model
+from db.models import Order, Ticket
+
+def create_order(tickets: list[dict], username: str, date: str = None) -> Order:
+    user = get_user_model().objects.get(username=username)
+
+    with transaction.atomic():
+        order = Order.objects.create(user=user)
+
+        if date:
+            order.created_at = date
+            order.save(update_fields=["created_at"])
+
+        for ticket in tickets:
+            Ticket.objects.create(
+                order=order,
+                movie_session_id=ticket["movie_session"],
+                row=ticket["row"],
+                seat=ticket["seat"]
+            )
+
+        return order

```

Comment:
Don't Repeat Yourself: The `save()` method is called twice in the `create_order` function. It's better to call it only once. You can set the `created_at` attribute and then save the order object. This will reduce the number of database queries. Here's how you can do it:

python
with transaction.atomic():
    order = Order.objects.create(user=user)
    if date:
        order.created_at = date
    order.save()
```

This way, the `save()` method is called only once, regardless of whether the `date` is provided or not.